### PR TITLE
added flipDirection/flipHour settings, stopname and filterDirection fix

### LINKED ISSUE #5: Add direction filter
See #2.

### PR DESCRIPTION
**Pull request by gmichaela**
Fixes #5.

added the filterDirection #5 filter, upon doing so, realized I care more about one direction in the morning and a different direction in the afternoon #6? -- added a few more config flags, if flipDirection is set to true it will switch the direction at noon, if you want an hour other than noon as your flip time -- override with flipHour.

few other minor edits/fixes also included.

### CODE REVIEW

File: MMM-MBTA.js [hunk 1]
```diff
@@ -235,7 +251,7 @@ Module.register("MMM-MBTA", {
                 if (this.config.showETATime) {
                     var preETACell = document.createElement("td");
                     var preETATime = this.stationData[i].preETA;
-
+
```
Comment (edward-shen) [inline MMM-MBTA.js]: Please remove trailing whitespace.
```suggestion

```

File: MMM-MBTA.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
 Module.register("MMM-MBTA", {
     defaults: {
         apikey: "",
-        updateInterval: 10, // In seconds
+        updateInterval: 30, // In seconds
```
Comment (edward-shen) [inline MMM-MBTA.js]: Any particular reason for the change?

File: MMM-MBTA.js [hunk 3]
```diff
@@ -20,8 +20,11 @@ Module.register("MMM-MBTA", {
         fadePoint: 0.25, // Start on 1/4th of the list.
         showFullName: false,
         colorIcons: false,
-        showAlerts: false,
-        hideEmptyAlerts: false
+        showAlerts: false,  // works but styling needs help
+        hideEmptyAlerts: false,
+        flipDirection: false,  // if set to true, it will flip direction filter flag,
+        flipHour: 12, 
+        directionFlipped: false
```
Comment (edward-shen) [inline MMM-MBTA.js:27]: For all these values, can you please update the readme with these flags? Following what's been done is good enough!

File: MMM-MBTA.js [hunk 4]
```diff
@@ -124,7 +128,7 @@ Module.register("MMM-MBTA", {
             row.appendChild(symbolCell);
 
             var descCell = document.createElement("td");
-            descCell.innerHTML = "No vehicles are scheduled";
+            descCell.innerHTML = "Nothing coming soon";
```
Comment (edward-shen) [inline MMM-MBTA.js]: Missing a period or a "..."

```suggestion
            descCell.innerHTML = "Nothing coming soon...";
```

File: MMM-MBTA.js [hunk 5]
```diff
@@ -436,9 +452,9 @@ Module.register("MMM-MBTA", {
         // Gets (maxEntries + 10) schedules or all schedules up to 3 hours from now, whichever is lower
         // Page and time limits necessary because otherwise "schedules" endpoint gets every single schedule
         if (!this.config.predictedTimes) {
-            url += "&page[limit]=" + (maxEntries + 10) + '"';
+            url += "&page[limit]=" + (this.config.maxEntries + 10) + '"';
```
Comment (edward-shen) [inline MMM-MBTA.js:456]: Nice catch!

File: MMM-MBTA.js [hunk 6]
```diff
@@ -436,9 +452,9 @@ Module.register("MMM-MBTA", {
...
             url += "&filter[min_time]=" + moment().format("HH:mm");
-            url += "&filter[max_time]=" + moment().add(3, 'h').format("HH:mm");
+            url += "&filter[max_time]=" + moment().add(5, 'h').format("HH:mm");
```
Comment (edward-shen) [inline MMM-MBTA.js:458]: Any particular reason for this change? if you want to keep it at five, make sure to update the comment on line 452!

File: MMM-MBTA.js [hunk 7]
```diff
@@ -613,6 +631,34 @@ Module.register("MMM-MBTA", {
         // Sorts them according to ETA time
         this.stationData.sort((a,b) => (a.preETA - b.preETA));
 
+        if (this.config.flipDirection && this.config.direction.length > 0)
+        {
```
Comment (edward-shen) [inline MMM-MBTA.js]: Please be consistent with the already existing guide. This should be

```
if (...) {
  ...
}
```

Thanks!

Comment (edward-shen) [reply MMM-MBTA.js]: There are a few more in the follow code too!

File: MMM-MBTA.js [hunk 8]
```diff
@@ -613,6 +631,34 @@ Module.register("MMM-MBTA", {
...
+            {
+                this.directionFlipped = false;
+            }
+
+        }
+
+        //  Applys directional filter
+        if (this.filterDirection.length > 0)
+        {
+            this.stationData = this.stationData.filter(obj => (obj.directionId == this.filterDirection));
```
Comment (edward-shen) [inline MMM-MBTA.js]: Double equals is discouraged. This should be `===`.
```suggestion
            this.stationData = this.stationData.filter((obj) => obj.directionId === this.filterDirection);
```

File: MMM-MBTA.js [hunk 9]
```diff
@@ -198,8 +202,20 @@ Module.register("MMM-MBTA", {
...
+                switch (this.stationData[i].directionId) {
+                    case "0":
+                        direction = " Out";
+                        break;
+                    case "1":
+                        direction = " In";
+                        break;
+                }
 
+
```
Comment (edward-shen) [inline MMM-MBTA.js]: Please leave only one blank lines between one thing and the next.

**Review by edward-shen — CHANGES_REQUESTED** [review]
Looks like a great PR! I just have some minor requests and then it looks good!

**Comment by edward-shen** [comment]
Fixes #5.

File: MMM-MBTA.js (hunk 6 above)
Comment (gmichaela) [reply MMM-MBTA.js:458]: I wanted to see more trips coming, commuter rails don't have too many options in the 3 hour window sometimes...   the predictedTimes api doesnt seem to return too far in advance either, so I was just experimenting with adding more time here -- I saw that you truncated the list at the end anyway...

Comment (edward-shen) [reply MMM-MBTA.js:458]: Alright, that's understandable! Just update the comment and you should be all set for this part!

**Comment by gmichaela** [comment]
Made some more edits -- updated readme, fixed a filter direction bug.

File: MMM-MBTA.js (hunk 8 above)
Comment (gmichaela) [reply MMM-MBTA.js]: ended up it should have been =

File: readme.md [hunk 10]
```diff
@@ -67,12 +67,17 @@ Option|Description
 `colorIcons`|Display the vehicle icons in their respective color.<br/>**Expected Value type:** `boolean`.
 `showAlerts`|Shows alerts near the specified station.<br/>**Expected Value type:** `boolean`.<br/>**WARNING**: Alerts are only properly shown when in a full-width position, such as `top_bar`, `middle_center` or `lower_third`.
 `hideEmptyAlerts`|Automatically hide the alerts section when there are no alerts.<br/>**Expected Value type:** `boolean`.
+`direction`| Sets the default direction you care about.<br/>Example: `direction:  "Inbound" `.<br/>**Expected Value type:** `String`.
+`flipDirection`| Set to true, it will flip direction filter flag, ie: Inbound in the AM Outbound in PM<br/>**Expected Value type:** `boolean`.
+`flipHour`| When set with flipDirection the time of day you want to flip direction to override 12 noon<br/>**Expected Value type:** `int`.
```
Comment (edward-shen) [inline readme.md]:
```suggestion
`flipHour`| Specifies the hour to flip the direction, if `flipDirection` is equal to `true`. Default value is 12.**Expected Value type:** `int`.
```

File: readme.md [hunk 11]
```diff
@@ -67,12 +67,17 @@ Option|Description
 `colorIcons`|Display the vehicle icons in their respective color.<br/>**Expected Value type:** `boolean`.
 `showAlerts`|Shows alerts near the specified station.<br/>**Expected Value type:** `boolean`.<br/>**WARNING**: Alerts are only properly shown when in a full-width position, such as `top_bar`, `middle_center` or `lower_third`.
 `hideEmptyAlerts`|Automatically hide the alerts section when there are no alerts.<br/>**Expected Value type:** `boolean`.
+`direction`| Sets the default direction you care about.<br/>Example: `direction:  "Inbound" `.<br/>**Expected Value type:** `String`.
+`flipDirection`| Set to true, it will flip direction filter flag, ie: Inbound in the AM Outbound in PM<br/>**Expected Value type:** `boolean`.
```
Comment (edward-shen) [inline readme.md]:
```suggestion
`flipDirection`| When set to `true`, MMM-MBTA will the flip the direction filter for incoming vehicles at the station, i.e. Inbound in the AM and Outbound in PM<br/>**Expected Value type:** `boolean`.
```

**Review by edward-shen — APPROVED** [review]
Please accept the suggestions and your PR will be merged! 

Thanks for working with me on this, I really appreciate the PR!

File: readme.md (hunk 10 above)
Comment (gmichaela) [reply readme.md]: Adding Outbound etc wouldn't hurt either...

**Comment by edward-shen** [comment]
Merged! :tada:

**Pull request merged**